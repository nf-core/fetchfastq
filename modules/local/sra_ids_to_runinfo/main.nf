
process SRA_IDS_TO_RUNINFO {
    tag "$id"
    label 'error_retry'

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    String id
    String fields

    output:
    Path tsv = path("*.runinfo.tsv")

    topic:
    [ task.process, 'python', eval("python --version | sed 's/Python //g'") ] >> 'versions'

    script:
    def metadata_fields = fields ? "--ena_metadata_fields ${fields}" : ''
    """
    echo $id > id.txt
    sra_ids_to_runinfo.py \\
        id.txt \\
        ${id}.runinfo.tsv \\
        $metadata_fields
    """
}
