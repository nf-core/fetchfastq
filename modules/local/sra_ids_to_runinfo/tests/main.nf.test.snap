{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo.tsv:md5,3a1be35781ca6e8a28d8fd4d2f3bbe85"
                ],
                "1": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ],
                "tsv": [
                    "SRR13191702.runinfo.tsv:md5,3a1be35781ca6e8a28d8fd4d2f3bbe85"
                ],
                "versions": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:17:11.040523"
    }
}